windows/src/developer/TIKE/project/Keyman.Developer.UI.Project.UfrmNewProject.pas

```diff
@@ -0,0 +1,189 @@
+      Selected: Boolean);
+  private
+    procedure EnableControls;
+    function GetProjectType: TKeymanProjectType;
+    procedure UpdateDescription;
+  protected
+    function GetHelpTopic: string; override;
+  public
+    property ProjectType: TKeymanProjectType read GetProjectType;
+  end;
+
+function ShowNewProjectForm(Owner: TComponent): Boolean;
+
+implementation
+
+uses
+  Keyman.Developer.System.HelpTopics,
+  Keyman.Developer.System.Project.Project,
+  Keyman.Developer.UI.Project.ProjectUI,
+  Keyman.Developer.UI.Project.UfrmNewProjectParameters,
+  Keyman.Developer.UI.ImportWindowsKeyboardDialogManager,
+  UfrmMain,
+  utilsystem;
+
+function ShowNewProjectForm(Owner: TComponent): Boolean;
+var
+  p: TfrmNewProject;
+  pt: TKeymanProjectType;
+begin
+  Result := False;
+
+  p := TfrmNewProject.Create(Owner);
+  try
+    if p.ShowModal = mrCancel then
+      Exit;
+    pt := p.ProjectType;
+  finally
+    p.Free;
+  end;
+
+  case pt of
+    kptUnknown:
+      Assert(False, 'Should not be able to have a project type of unknown here');
+    kptBasic:
+      Result := ShowNewProjectParameters(Owner);
+    kptBlank:
+      begin
+        FGlobalProject.Save;
+        frmKeymanDeveloper.ProjectForm.Free;
+        FreeGlobalProjectUI;
+        LoadGlobalProjectUI('');
+        frmKeymanDeveloper.ShowProject;
+        Result := True;
+      end;
+    kptImportWindowsKeyboard:
+      Result := ShowImportWindowsKeyboard(Owner);
+  end;
+end;
+
+{$R *.DFM}
+
+procedure TfrmNewProject.EnableControls;
+var
+  e: Boolean;
+begin
+  e := Assigned(lvItems.Selected);
+  cmdOK.Enabled := e;
+end;
+
+procedure TfrmNewProject.FormCreate(Sender: TObject);
+begin
+  inherited;
+  lvItems.Selected := lvItems.Items[0];
+  lvItems.ItemFocused := lvItems.Items[0];
+
+  UpdateDescription;
+  EnableControls;
+end;
+
+function TfrmNewProject.GetProjectType: TKeymanProjectType;
+begin
+  // Overly wordy but clarifies that we may have other types later
+  if not Assigned(lvItems.Selected) then
+    Result := kptUnknown
+  else if lvItems.Selected.Caption = 'Basic' then
+    Result := kptBasic
+  else if lvItems.Selected.Caption = 'Blank' then
+    Result := kptBlank
+  else if lvItems.Selected.Caption = 'Import Windows Keyboard' then
+    Result := kptImportWindowsKeyboard
+  else
+    Result := kptUnknown;
+end;
+
+function TfrmNewProject.GetHelpTopic: string;
+begin
+  Result := SHelpTopic_Context_New;
+end;
+
+procedure TfrmNewProject.lvItemsDblClick(Sender: TObject);
+begin
+  if cmdOK.Enabled then
+    cmdOK.Click;
+end;
+
+procedure TfrmNewProject.lvItemsSelectItem(Sender: TObject; Item: TListItem;
+  Selected: Boolean);
+begin
+  UpdateDescription;
+  EnableControls;
+end;
+
+procedure TfrmNewProject.UpdateDescription;
+begin
+  case ProjectType of
+    kptUnknown: lblDescription.Caption := '';
+    kptBasic: lblDescription.Caption :=
+      'Creates a keyboard project that matches the layout used in '+
```

Comment:
Since "layout" is usually describing keyboard layouts, maybe use a different description like "folder structure"?

Reply:
Good point.